src/sentry/static/sentry/app/components/buttons/button.jsx

```diff
@@ -90,70 +74,170 @@ class Button extends React.Component {
-      'button-default': !isLink && !isPrimary && !isDanger,
-      'button-zero': size === 'zero',
-      'button-sm': size === 'small',
-      'button-xs': size === 'xsmall',
-      'button-lg': size === 'large',
-      'button-busy': busy,
-      'button-disabled': disabled,
     });
 
-    let childContainer = (
-      <Flex align="center" className="button-label">
-        {icon && (
-          <Icon size={size}>
-            <StyledInlineSvg src={icon} size={size === 'small' ? '12px' : '16px'} />
-          </Icon>
-        )}
-        {children}
-      </Flex>
-    );
-
     // Buttons come in 3 flavors: Link, anchor, and regular buttons. Let's
     // use props to determine which to serve up, so we don't have to think
     // about it. As a bonus, let's ensure all buttons appear as a button
     // control to screen readers. Note: you must still handle tabindex manually.
 
-    // Props common to all elements
-    let componentProps = {
-      disabled,
-      ...buttonProps,
-      onClick: this.handleClick,
-      className: cx,
-      role: 'button',
-      children: childContainer,
-    };
-
-    // Handle react-router Links
-    if (to) {
-      return <Link to={this.getUrl()} {...componentProps} />;
-    }
+    return (
+      <StyledSmartButton
+        to={this.getUrl(to)}
+        href={this.getUrl(href)}
+        size={size}
+        {...buttonProps}
+        onClick={this.handleClick}
+        className={classNames(cxs)}
+        role="button"
+      >
+        <ButtonLabel size={size}>
+          {icon && (
+            <Icon size={size} hasChildren={!!children}>
+              <StyledInlineSvg src={icon} size={size === 'small' ? '12px' : '16px'} />
+            </Icon>
+          )}
+          {children}
+        </ButtonLabel>
+      </StyledSmartButton>
+    );
+  }
+}
 
-    if (href && external) {
-      return <ExternalLink href={this.getUrl()} {...componentProps} />;
-    }
+export default Button;
+
+const getFontSize = ({size}) => {
+  switch (size) {
+    case 'xsmall':
+    case 'small':
+      return '12px';
+    case 'large':
+      return '16px';
+    default:
+      return '14px';
+  }
+};
+
+const getFontWeight = ({priority}) => `font-weight: ${priority === 'link' ? 400 : 600};`;
+
+const getBoxShadow = active => ({priority, borderless, disabled}) => {
+  if (disabled || borderless || priority === 'link') {
+    return 'box-shadow: none';
+  }
 
-    // Handle traditional links
-    if (href) {
-      return <a href={this.getUrl()} {...componentProps} />;
+  return `box-shadow: ${active ? 'inset' : ''} 0 2px rgba(0, 0, 0, 0.05)`;
+};
+
+const getColors = ({priority, disabled, theme}) => {
+  let themeName = disabled ? 'disabled' : priority || 'default';
+  let {
+    color,
+    colorActive,
+    background,
+    backgroundActive,
+    border,
+    borderActive,
+  } = theme.button[themeName];
+
+  return css`
+    color: ${color};
+    background: ${background};
```

Comment:
let's use `background-color` here and in the active state since this is a big deal and it adds just a bit of flexibility (`background` overrides other properties)